### ISSUE TITLE
[11.0]

### ISSUE THREAD

**Issue by WeeDom** [OCA/OpenUpgrade#2026]
Impacted versions: 11.0

Steps to reproduce: 

1. clone openupgrade.git
2. checkout 11.0 branch
3. make dockerfile with 10.0 database mounted and odoo-10+e.deb installed
4. docker-compose mounts openupgrade repo to /home/openupgrade/ in container
5. odoo.conf addons_path has /usr/lib/python2.7/dist-packages/odoo/addons and /home/openupgrade/addons
6. run odoo@246298db0b67:/home/openupgrade$ ./odoo-bin --update all --stop-after-init --dev=all -d miningsearchesuk

 Current behavior:
```
2019-09-18 13:49:34,946 53 DEBUG ? odoo.netsvc: logger level set: "odoo.http.rpc.request:INFO"
2019-09-18 13:49:34,946 53 DEBUG ? odoo.netsvc: logger level set: "odoo.http.rpc.response:INFO"
2019-09-18 13:49:34,947 53 DEBUG ? odoo.netsvc: logger level set: ":INFO"
2019-09-18 13:49:34,947 53 DEBUG ? odoo.netsvc: logger level set: "werkzeug:CRITICAL"
2019-09-18 13:49:34,947 53 DEBUG ? odoo.netsvc: logger level set: ":DEBUG"
2019-09-18 13:49:34,947 53 INFO ? odoo: Odoo version 11.0
2019-09-18 13:49:34,947 53 INFO ? odoo: Using configuration file at /etc/odoo/odoo.conf
2019-09-18 13:49:34,947 53 INFO ? odoo: addons paths: ['/mnt/extra-addons', '/usr/lib/python2.7/dist-packages/odoo/addons', '/home/openupgrade/addons', '/home/openupgrade/odoo/addons']
2019-09-18 13:49:34,948 53 INFO ? odoo: database: odoo@msuk_db:default
2019-09-18 13:49:34,948 53 DEBUG ? odoo.sql_db: create serialized cursor to {'database': 'postgres', 'host': 'msuk_db', 'user': 'odoo', 'password': '', 'sslmode': 'prefer'}
2019-09-18 13:49:34,952 53 DEBUG ? odoo.sql_db: ConnectionPool(used=1/count=1/max=64) Create new connection
2019-09-18 13:49:34,953 53 DEBUG ? odoo.sql_db: query: SELECT datname FROM pg_database WHERE datname = %s
2019-09-18 13:49:34,955 53 DEBUG ? odoo.sql_db: SQL LOG from:
2019-09-18 13:49:34,955 53 DEBUG ? odoo.sql_db: table: pg_database: 0:00:00.001820/1
2019-09-18 13:49:34,955 53 DEBUG ? odoo.sql_db: SUM from:0:00:00.001820/1 [1]
2019-09-18 13:49:34,955 53 DEBUG ? odoo.sql_db: SUM into:0:00:00/1 [1]
2019-09-18 13:49:34,956 53 DEBUG ? odoo.sql_db: ConnectionPool(used=1/count=1/max=64) Give back connection to 'user=odoo password=xxx dbname=postgres host=msuk_db sslmode=prefer'
2019-09-18 13:49:34,956 53 DEBUG ? odoo.sql_db: ConnectionPool(used=0/count=0/max=64) Forgot connection to 'user=odoo password=xxx dbname=postgres host=msuk_db sslmode=prefer'
2019-09-18 13:49:34,959 53 CRITICAL ? odoo.modules.module: Couldn't load module web
2019-09-18 13:49:34,959 53 CRITICAL ? odoo.modules.module: No module named 'controllers'
2019-09-18 13:49:34,959 53 ERROR ? odoo.service.server: Failed to load server-wide module `web`.
The `web` module is provided by the addons found in the `openerp-web` project.
Maybe you forgot to add those addons in your addons_path configuration.
Traceback (most recent call last):
  File "/home/openupgrade/odoo/service/server.py", line 947, in load_server_wide_modules
    odoo.modules.module.load_openerp_module(m)
  File "/home/openupgrade/odoo/modules/module.py", line 368, in load_openerp_module
    __import__('odoo.addons.' + module_name)
  File "/home/openupgrade/odoo/modules/module.py", line 82, in load_module
    exec(open(modfile, 'rb').read(), new_mod.__dict__)
  File "<string>", line 4, in <module>
ModuleNotFoundError: No module named 'controllers'
2019-09-18 13:49:34,965 53 WARNING ? odoo.service.server: 'inotify' module not installed. Code autoreload feature is disabled
2019-09-18 13:49:34,966 53 DEBUG ? odoo.service.server: Setting signal handlers
2019-09-18 13:49:34,966 53 DEBUG ? odoo.sql_db: create serialized cursor to {'database': 'miningsearchesuk', 'host': 'msuk_db', 'user': 'odoo', 'password': '', 'sslmode': 'prefer'}
2019-09-18 13:49:34,969 53 DEBUG ? odoo.sql_db: ConnectionPool(used=1/count=1/max=64) Create new connection
2019-09-18 13:49:34,970 53 DEBUG ? odoo.sql_db: query: SELECT proname FROM pg_proc WHERE proname='unaccent'
2019-09-18 13:49:34,972 53 DEBUG ? odoo.sql_db: SQL LOG from:
2019-09-18 13:49:34,972 53 DEBUG ? odoo.sql_db: table: pg_proc: 0:00:00.001782/1
2019-09-18 13:49:34,972 53 DEBUG ? odoo.sql_db: SUM from:0:00:00.001782/1 [2]
2019-09-18 13:49:34,972 53 DEBUG ? odoo.sql_db: SUM into:0:00:00/1 [2]
2019-09-18 13:49:34,972 53 DEBUG ? odoo.sql_db: ConnectionPool(used=1/count=1/max=64) Give back connection to 'user=odoo password=xxx dbname=miningsearchesuk host=msuk_db sslmode=prefer'
2019-09-18 13:49:34,972 53 DEBUG ? odoo.sql_db: ConnectionPool(used=0/count=1/max=64) Put connection to 'user=odoo password=xxx dbname=miningsearchesuk host=msuk_db sslmode=prefer' in pool
2019-09-18 13:49:34,973 53 DEBUG ? odoo.sql_db: create serialized cursor to {'database': 'miningsearchesuk', 'host': 'msuk_db', 'user': 'odoo', 'password': '', 'sslmode': 'prefer'}
2019-09-18 13:49:34,973 53 DEBUG ? odoo.sql_db: ConnectionPool(used=1/count=1/max=64) Borrow existing connection to 'user=odoo password=xxx dbname=miningsearchesuk host=msuk_db sslmode=prefer' at index 0
2019-09-18 13:49:34,973 53 DEBUG ? odoo.sql_db: query: SELECT sequence_name FROM information_schema.sequences WHERE sequence_name='base_registry_signaling'
2019-09-18 13:49:34,976 53 DEBUG ? odoo.sql_db: query:  SELECT base_registry_signaling.last_value,
                                  base_cache_signaling.last_value
                           FROM base_registry_signaling, base_cache_signaling
2019-09-18 13:49:34,977 53 DEBUG ? odoo.modules.registry: Multiprocess load registry signaling: [Registry: 7] [Cache: 8817]
2019-09-18 13:49:34,978 53 DEBUG ? odoo.sql_db: SUM from:0:00:00/2 [4]
2019-09-18 13:49:34,978 53 DEBUG ? odoo.sql_db: SUM into:0:00:00/2 [4]
2019-09-18 13:49:34,978 53 DEBUG ? odoo.sql_db: ConnectionPool(used=1/count=1/max=64) Give back connection to 'user=odoo password=xxx dbname=miningsearchesuk host=msuk_db sslmode=prefer'
2019-09-18 13:49:34,978 53 DEBUG ? odoo.sql_db: ConnectionPool(used=0/count=1/max=64) Put connection to 'user=odoo password=xxx dbname=miningsearchesuk host=msuk_db sslmode=prefer' in pool
2019-09-18 13:49:34,978 53 DEBUG ? odoo.sql_db: create serialized cursor to {'database': 'miningsearchesuk', 'host': 'msuk_db', 'user': 'odoo', 'password': '', 'sslmode': 'prefer'}
2019-09-18 13:49:34,979 53 DEBUG ? odoo.sql_db: ConnectionPool(used=1/count=1/max=64) Borrow existing connection to 'user=odoo password=xxx dbname=miningsearchesuk host=msuk_db sslmode=prefer' at index 0
2019-09-18 13:49:34,979 53 DEBUG ? odoo.sql_db: query:
        SELECT c.relname
          FROM pg_class c
          JOIN pg_namespace n ON (n.oid = c.relnamespace)
         WHERE c.relname IN %s
           AND c.relkind IN ('r', 'v', 'm')
           AND n.nspname = 'public'

2019-09-18 13:49:34,981 53 DEBUG miningsearchesuk odoo.sql_db: query: update ir_module_module set state=%s where name=%s and state=%s
2019-09-18 13:49:34,983 53 DEBUG miningsearchesuk odoo.sql_db: query: SELECT name, id, state, demo AS dbdemo, latest_version AS installed_version  FROM ir_module_module WHERE name IN %s
2019-09-18 13:49:34,985 53 INFO miningsearchesuk odoo.modules.loading: loading 1 modules...
2019-09-18 13:49:34,986 53 DEBUG miningsearchesuk odoo.modules.loading: loading module base (1/1)
2019-09-18 13:49:34,987 53 CRITICAL miningsearchesuk odoo.modules.module: Couldn't load module base
2019-09-18 13:49:34,987 53 CRITICAL miningsearchesuk odoo.modules.module: No module named 'ir'
2019-09-18 13:49:34,987 53 DEBUG miningsearchesuk odoo.sql_db: SQL LOG from:
2019-09-18 13:49:34,987 53 DEBUG miningsearchesuk odoo.sql_db: table: ir_module_module: 0:00:00.000716/1
2019-09-18 13:49:34,988 53 DEBUG miningsearchesuk odoo.sql_db: SUM from:0:00:00.000716/3 [7]
2019-09-18 13:49:34,988 53 DEBUG miningsearchesuk odoo.sql_db: SUM into:0:00:00/3 [7]
2019-09-18 13:49:34,988 53 DEBUG miningsearchesuk odoo.sql_db: ConnectionPool(used=1/count=1/max=64) Give back connection to 'user=odoo password=xxx dbname=miningsearchesuk host=msuk_db sslmode=prefer'
2019-09-18 13:49:34,988 53 DEBUG miningsearchesuk odoo.sql_db: ConnectionPool(used=0/count=1/max=64) Put connection to 'user=odoo password=xxx dbname=miningsearchesuk host=msuk_db sslmode=prefer' in pool
2019-09-18 13:49:34,989 53 DEBUG miningsearchesuk odoo.sql_db: create serialized cursor to {'database': 'miningsearchesuk', 'host': 'msuk_db', 'user': 'odoo', 'password': '', 'sslmode': 'prefer'}
2019-09-18 13:49:34,990 53 DEBUG miningsearchesuk odoo.sql_db: ConnectionPool(used=1/count=1/max=64) Borrow existing connection to 'user=odoo password=xxx dbname=miningsearchesuk host=msuk_db sslmode=prefer' at index 0
2019-09-18 13:49:34,990 53 DEBUG miningsearchesuk odoo.sql_db: query: UPDATE ir_module_module SET state='installed' WHERE state IN ('to remove', 'to upgrade')
2019-09-18 13:49:34,992 53 DEBUG miningsearchesuk odoo.sql_db: query: UPDATE ir_module_module SET state='uninstalled' WHERE state='to install'
2019-09-18 13:49:34,993 53 WARNING miningsearchesuk odoo.modules.loading: Transient module states were reset
2019-09-18 13:49:34,993 53 DEBUG miningsearchesuk odoo.sql_db: SUM from:0:00:00/2 [9]
2019-09-18 13:49:34,993 53 DEBUG miningsearchesuk odoo.sql_db: SUM into:0:00:00/2 [9]
2019-09-18 13:49:34,994 53 DEBUG miningsearchesuk odoo.sql_db: ConnectionPool(used=1/count=1/max=64) Give back connection to 'user=odoo password=xxx dbname=miningsearchesuk host=msuk_db sslmode=prefer'
2019-09-18 13:49:34,994 53 DEBUG miningsearchesuk odoo.sql_db: ConnectionPool(used=0/count=1/max=64) Put connection to 'user=odoo password=xxx dbname=miningsearchesuk host=msuk_db sslmode=prefer' in pool
2019-09-18 13:49:34,994 53 ERROR miningsearchesuk odoo.modules.registry: Failed to load registry
Traceback (most recent call last):
  File "/home/openupgrade/odoo/modules/registry.py", line 85, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/home/openupgrade/odoo/modules/loading.py", line 385, in load_modules
    report=report, models_to_check=models_to_check, upg_registry=upg_registry)
  File "/home/openupgrade/odoo/modules/loading.py", line 153, in load_module_graph
    load_openerp_module(package.name)
  File "/home/openupgrade/odoo/modules/module.py", line 368, in load_openerp_module
    __import__('odoo.addons.' + module_name)
  File "/home/openupgrade/odoo/modules/module.py", line 82, in load_module
    exec(open(modfile, 'rb').read(), new_mod.__dict__)
  File "<string>", line 4, in <module>
ModuleNotFoundError: No module named 'ir'
2019-09-18 13:49:34,995 53 CRITICAL miningsearchesuk odoo.service.server: Failed to initialize database `miningsearchesuk`.
Traceback (most recent call last):
  File "/home/openupgrade/odoo/service/server.py", line 1006, in preload_registries
    registry = Registry.new(dbname, update_module=update_module)
  File "/home/openupgrade/odoo/modules/registry.py", line 85, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/home/openupgrade/odoo/modules/loading.py", line 385, in load_modules
    report=report, models_to_check=models_to_check, upg_registry=upg_registry)
  File "/home/openupgrade/odoo/modules/loading.py", line 153, in load_module_graph
    load_openerp_module(package.name)
  File "/home/openupgrade/odoo/modules/module.py", line 368, in load_openerp_module
    __import__('odoo.addons.' + module_name)
  File "/home/openupgrade/odoo/modules/module.py", line 82, in load_module
    exec(open(modfile, 'rb').read(), new_mod.__dict__)
  File "<string>", line 4, in <module>
ModuleNotFoundError: No module named 'ir'
```
Expected behavior:
upgrade db to v11

any help appreciated.

Thanks in advance

**Comment by pedrobaeza** [comment]
Your deployment or previous version DB is not correct, claiming there's a missing `ir` module. Maybe an incorrect addons path. Any way, this is not something of OpenUpgrade itself, so closing.

**Comment by WeeDom** [comment]
for future travellers, I fixed it. Mostly - got a different error, but got past this one.

The problem actually lies in my addons path. See towards the top of the stack:

`'/usr/lib/python2.7/dist-packages/odoo/addons'`

So it was trying to find the odoo 10 addons - which were written in python2.7 - and compile them using python3 which openupgrade requires. 

I took that out of the addons, and now get a different error message... but this current problem is solved.
